lib/galaxy/web/base/controller.py

```diff
@@ -280,7 +280,10 @@ def client(self, trans, **kwd):
         Should not be used with url_for -- see
         (https://github.com/galaxyproject/galaxy/issues/1878) for why.
         """
-        self._check_require_login(trans)
+        if kwd.get("successful_oidc_callback", False):
```

Comment:
I haven't had a chance to set this up and tinker with it locally, but it doesn't seem right to push this logic into the default client endpoint.  I can try a bit later to see what I can come up with, if you'd like.

Reply:
We can push it `authnz.callback`, but I remember we had issues with that previously. Regardless, please feel free to modify if you see a more appropriate solution.